cassandra-sink/src/main/java/org/springframework/cloud/stream/module/cassandra/sink/CassandraSinkConfiguration.java

```diff
@@ -0,0 +1,205 @@
+import org.springframework.integration.handler.AbstractMessageProducingHandler;
+import org.springframework.integration.handler.BridgeHandler;
+import org.springframework.integration.support.json.Jackson2JsonObjectMapper;
+import org.springframework.integration.support.json.JsonObjectMapper;
+import org.springframework.integration.transformer.AbstractPayloadTransformer;
+import org.springframework.integration.transformer.MessageTransformingHandler;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.util.StringUtils;
+import org.springframework.cloud.stream.module.cassandra.CassandraConfiguration;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.util.StdDateFormat;
+
+/**
+ * @author Artem Bilan
+ * @author Thomas Risberg
+ */
+@Configuration
+@Import(CassandraConfiguration.class)
+@EnableIntegration
+@EnableConfigurationProperties(CassandraSinkProperties.class)
+public class CassandraSinkConfiguration {
+
+	private static final SpelExpressionParser PARSER = new SpelExpressionParser();
+
+	@Autowired
+	private CassandraSinkProperties cassandraSinkProperties;
+	
+	@Autowired
+	private CassandraOperations template;
+
+	@Bean
+	public MessageChannel input() {
+		return new DirectChannel();
+	}
+
+	@Bean
+	public MessageChannel toSink() {
+		return new DirectChannel();
+	}
+
+	@Bean
+	@Primary
+	public MessageHandler bridgeMessageHandler() {
+		AbstractMessageProducingHandler messageHandler;
+		if (StringUtils.hasText(cassandraSinkProperties.getIngestQuery())) {
+			messageHandler = new MessageTransformingHandler(new PayloadToMatrixTransformer(cassandraSinkProperties.getIngestQuery()));
+		}
+		else {
+			messageHandler = new BridgeHandler();
+		}
+		messageHandler.setOutputChannel(toSink());
+		return messageHandler;
+	}
+
+	@Bean
+	@ServiceActivator(inputChannel = "toSink")
+	public MessageHandler cassandraSinkMessageHandler() {
+		CassandraMessageHandler<?> cassandraMessageHandler = new CassandraMessageHandler<>(this.template, cassandraSinkProperties.getQueryType());
+		cassandraMessageHandler.setProducesReply(false);
+		if (cassandraSinkProperties.getConsistencyLevel() != null || cassandraSinkProperties.getRetryPolicy() != null || cassandraSinkProperties.getTtl() > 0) {
+			cassandraMessageHandler.setWriteOptions(
+					new WriteOptions(cassandraSinkProperties.getConsistencyLevel(), cassandraSinkProperties.getRetryPolicy(), cassandraSinkProperties.getTtl()));
+		}
+		if (StringUtils.hasText(cassandraSinkProperties.getIngestQuery())) {
+			cassandraMessageHandler.setIngestQuery(cassandraSinkProperties.getIngestQuery());
+		}
+		else if (StringUtils.hasText(cassandraSinkProperties.getStatementExpression())) {
+			Expression expression = PARSER.parseExpression(cassandraSinkProperties.getStatementExpression());
```

Comment:
Didn't we find the compromise with this `Expression` stuff ?
At least, shouldn't we follow here with the `FileSinkConfiguration` experience ?